crates/matrix-sdk-crypto/src/identities/manager.rs

```diff
@@ -108,17 +107,8 @@ impl KeysQueryListener {
     ///
     /// If the given timeout has elapsed the method will stop waiting and return
     /// an error.
-    pub async fn wait(&self, timeout: Duration) -> Result<(), Elapsed> {
-        let listener = self.inner.listen();
-
-        #[cfg(not(target_arch = "wasm32"))]
-        tokio::time::timeout(timeout, async { listener.await }).await.map_err(|_| Elapsed(()))?;
-
-        // TODO we should ensure that this is async on wasm as well.
-        #[cfg(target_arch = "wasm32")]
-        listener.wait_timeout(timeout);
-
-        Ok(())
+    pub async fn wait(&self, timeout: Duration) -> Result<(), ElapsedError> {
+        wait(async { self.inner.listen().await }, timeout).await
```

Comment:
I think this async block is a no-op.

```suggestion
        wait(self.inner.listen(), timeout).await
```